config.yaml

```diff
@@ -2,18 +2,31 @@
 # See LICENSE file for licensing details.
 
 options:
-  git-sync::repo:
+  git_repo:
     type: string
-    description: URL to repo to sync against
-  git-sync::branch:
+    description: URL to repo to clone and sync against.
+  git_branch:
```

Comment:
git_reference

Reply:
git_branch and git_rev are both defined and both needed. git-sync has defaults for both so we're good. Can provide `git_rev` and omit the branch. It's impossible to clone a revision - only after cloning a branch can checkout a revision.